File: task/services/analytics/app/schemas/stages.py
"""Pipeline stage enum and name mapping."""

from enum import IntEnum


class Stage(IntEnum):
    """1-based to match the API and legacy portal (ADR 0003)."""

    APPLIED = 1
    SCREENING = 2
    INTERVIEW = 3
    OFFER = 4
    HIRED = 5


STAGE_NAMES: dict[int, str] = {stage.value: stage.name.lower() for stage in Stage}

File: task/packages/contracts/src/stages.ts
// Stage indexes are 1-based in every API response and event payload,
// see docs/adr/0003; index 0 is never used. The legacy HR portal treats
// stage 0 as "missing".
export const STAGES = [
  'applied',
  'screening',
  'interview',
  'offer',
  'hired',
] as const;

export type StageName = (typeof STAGES)[number];

export const STAGE_COUNT = 5;
